qiita_db/sql_connection.py

```diff
@@ -460,176 +397,199 @@ def execute_fetchall(self, sql, sql_args=None):
 
         return result
 
-    def _check_queue_exists(self, queue_name):
-        """Checks if queue `queue_name` exists in the handler
 
-        Parameters
-        ----------
-        queue_name : str
-            The name of the queue
+class Transaction(object):
+    """Encapsulates a DB transaction
 
-        Returns
-        -------
-        bool
-            True if queue `queue_name` exist in the handler. False otherwise.
-        """
-        return queue_name in self.queues
+    A transaction is defined by a series of consecutive queries that need to
+    be applied to the database as a single block.
+
+    Parameters
+    ----------
+    name : str
+        Name of the transaction.
+    """
 
-    def create_queue(self, queue_name):
-        """Add a new queue to the connection
+    _regex = re.compile("^{(\d+):(\d+):(\d+)}$")
 
-        Parameters
-        ----------
-        queue_name : str
-            Name of the new queue
+    def __init__(self, name):
+        # The name is useful for debugging, since we can identify the
+        # failed queue in errors
+        self._name = name
+        self._queries = []
+        self._results = []
```

Comment:
Nope, originally was not storing the results of the last query. Originally was flattening the results of all queries and storing them in the list until something was used. If anything was used, the list was wiped out, but if not, it was stored everything in memory.

Again, if you are returning something from the database, you're going to have that memory footprint in the system anyways. If you think that the list is containing something that you don't need, probably you should re-visit your SQL queries.

You'll have the DictCursor anyway when you're returning a value from the database (i.e. executing a fetchall). Note that the only real overhead that this is adding is the 'None' values when the query is not returning anything. This value can be avoided using a dict instead of a list, however you then have an overhead on solving the indirections of the dictionary, so probably better as it is.